src/extensibility/node/ExtensionManagerDomain.js

```diff
@@ -98,17 +147,40 @@ function _performInstall(packagePath, installDirectory, validationResult, callba
         var sourceDir = path.join(validationResult.extractDir, validationResult.commonPrefix);
 
         fs.copy(sourceDir, installDirectory, function (err) {
-            if (err) {
+
+            var fail = function (err) {
                 _removeFailedInstallation(installDirectory);
                 callback(err, null);
-            } else {
+            };
+
+            var finish = function () {
                 // The status may have already been set previously (as in the
                 // DISABLED case.
                 if (!validationResult.installationStatus) {
                     validationResult.installationStatus = Statuses.INSTALLED;
                 }
                 callback(null, validationResult);
+            };
+
+            if (err) {
+                return fail(err);
+            }
+
+            var packageJson;
+
+            try {
+                packageJson = fs.readJsonSync(path.join(installDirectory, "package.json"));
+            } catch (e) {
+                packageJson = null;
             }
+
+            if (!packageJson || !packageJson.dependencies) {
+                return finish();
+            }
+
+            _performNpmInstall(installDirectory, function (err) {
+                return err ? fail(err) : finish();
```

Comment:
I think something like this would make easier to add / review code if we ever need it.
```js
if (err) {
  return fail(err);
}

finish();
```